miio/viomivacuum.py

```diff
@@ -277,46 +281,216 @@ def mop_mode(self) -> ViomiMode:
         """
         return ViomiMode(self.data["is_mop"])
 
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self.data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self.data["hw_info"]
+
+    @property
+    def sw_info(self) -> str:
+        """SoftWare info."""
+        return self.data["sw_info"]
+
+    @property
+    def hepa_hours_left(self) -> timedelta:
+        """HYPA left hours."""
+        return timedelta(self.data["hypa_hours"])
+
+    @property
+    def hepa_life_left(self) -> int:
+        """HYPA left life percent."""
+        return self.data["hypa_life"]
+
+    @property
+    def charging(self) -> bool:
+        """FIXME: True if device is charging?"""
```

Comment:
Could you try to verify the `FIXME` marked items?

Reply:
Yes ! few more props to test :)

Reply:
done